### ISSUE TITLE
TypeError: Cannot set property 'prev' of undefined

### ISSUE THREAD

**Issue by substack** [chriso/lru#19]
```
/home/substack/projects/kd-tree-store/node_modules/lru/index.js:33
            this.cache[this.tail].prev = null;
                                       ^

TypeError: Cannot set property 'prev' of undefined
    at LRU.remove (/home/substack/projects/kd-tree-store/node_modules/lru/index.js:33:40)
    at LRU.evict (/home/substack/projects/kd-tree-store/node_modules/lru/index.js:121:29)
    at LRU.set (/home/substack/projects/kd-tree-store/node_modules/lru/index.js:66:18)
    at build (/home/substack/projects/kd-tree-store/index.js:35:17)
    at write (/home/substack/projects/kd-tree-store/index.js:53:9)
    at build (/home/substack/projects/kd-tree-store/index.js:42:7)
    at write (/home/substack/projects/kd-tree-store/index.js:59:9)
    at /home/substack/projects/kd-tree-store/index.js:38:14
    at FSReqWrap.wrapper [as oncomplete] (fs.js:667:5)
```

**Comment by mafintosh** [comment]
i think this is fixed in latest version of LRU. do you know which version you are running?

**Comment by substack** [comment]
```
$ json version < node_modules/lru/package.json 
2.0.0
```

**Comment by mafintosh** [comment]
hmm interesting. is this error reproducible somehow?

**Comment by substack** [comment]
I haven't narrowed it down very much, but this code reproduces the error: https://gist.github.com/substack/536029236734be4431d2

**Comment by mafintosh** [comment]
should be fixed in 2.0.1

**Comment by mafintosh** [comment]
the crash was caused by the key `0` which we didn't coerce into a string which made it falsy
